staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -992,16 +992,29 @@ func (c completedConfig) New(name string, delegationTarget DelegationTarget) (*G
 func BuildHandlerChainWithStorageVersionPrecondition(apiHandler http.Handler, c *Config) http.Handler {
 	// WithStorageVersionPrecondition needs the WithRequestInfo to run first
 	handler := genericapifilters.WithStorageVersionPrecondition(apiHandler, c.StorageVersionManager, c.Serializer)
-	return DefaultBuildHandlerChain(handler, c)
+	return DefaultBuildHandlerChainFromAuthz(handler, c)
 }
 
+// DefaultBuildHandlerChain builds the default handler chain for the API server.
+// Is split in 2 parts - before and after authorization.
 func DefaultBuildHandlerChain(apiHandler http.Handler, c *Config) http.Handler {
-	handler := apiHandler
+	handler := DefaultBuildHandlerChainFromAuthz(apiHandler, c)
+	handler = DefaultBuildHandlerChainBeforeAuthz(handler, c)
+	return handler
+}
 
+// DefaultBuildHandlerChainFromAuthz builds the handler chain from authorization onwards.
+func DefaultBuildHandlerChainFromAuthz(apiHandler http.Handler, c *Config) http.Handler {
+	handler := apiHandler
 	handler = filterlatency.TrackCompleted(handler)
-	handler = genericapifilters.WithAuthorization(handler, c.Authorization.Authorizer, c.Serializer)
```

Comment:
why the order change? does it matter?

Reply:
right, old order was correct. We want to measure how long authz takes.

Reply:
This handler adding for delegations order not clicking in my head every time :D fixed (I hope so)

Reply:
> We want to measure how long authz takes

I thought that was already instrumented. If not, instrumenting seems useful and possible without splitting the handler chain, right?

Reply:
this was mistake in my ordering (handler load order,) fixed now, so noop)